pre_commit_hooks/byte_order_marker.py

```diff
@@ -0,0 +1,51 @@
+import os, sys, codecs
+import argparse
+from typing import Optional
+from typing import Sequence
+
+
+def main(argv: Optional[Sequence[str]] = None) -> int:
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '-f', '--fix',
+        choices=('auto', 'no'),
+        default='auto',
+        help='Remove the byte-order marker. Default is "auto"',
+    )
+    parser.add_argument('filenames', nargs='*', help='Filenames to check')
+    args = parser.parse_args(argv)
+
+    retv = 0
+
+    for filename in args.filenames:
+        with open(filename, 'rb') as f:
+            if f.read(3) == b'\xef\xbb\xbf':
+                if args.fix == 'no':
+                    print(f'{filename}: Found byte-order marker')
+                else:
+                    remove_bom_inplace(filename)
+                    print(f'{filename}: Removed byte-order marker')
+                retv = 1
+
+    return retv
+
+def remove_bom_inplace(path):
```

Comment:
sorry we can't take this code because it is not properly licensed, also it is unnecessarily complicated

an easier way to do this is:

```python
with open(..., newline='', encoding='utf-8-sig') as f:
    contents =f.read()

with open(..., newline='', encoding='utf-8') as f:
    f.write(contents)
```

Reply:
oh, you have to actually fill in the `...`s 🤦

Reply:
Yeah... not enough coffee at that point